astropy/table/table.py

```diff
@@ -1244,18 +1245,30 @@ def __setitem__(self, item, value):
             name = item
             # If this is a column-like object that could be added directly to table
             if isinstance(value, BaseColumn) or self._add_as_mixin_column(value):
+                # If we're setting a new column to a scalar, broadcast it.
+                # (things will fail in _init_from_cols if this doesn't work)
+                if (len(self) > 0 and (getattr(value, 'isscalar', False) or
+                                       getattr(value, 'shape', None) == () or
+                                       len(value) == 1)):
```

Comment:
This behavior of numpy (and hence Column) actually surprises me:
```
In [4]: x = np.arange(2)
In [5]: x[:] = [1]
In [6]: x
Out[6]: array([1, 1])
```
I personally think broadcasting a length=1 array to length=2 should fail in the same way that length=3 to length=2 fails.  It seems like a special case that a shape=(1,) array is degenerate with a scalar shape=(), whereas it is not degenerate with a shape=(1,1) array.  But maybe I'm just not seeing it right.

Anyway, that is what it is, so now I think I understand the `len(value)=1` here.